### PR TITLE
Lpd 25421

### PR DESCRIPTION
**Pull request by SamZiemer**
https://liferay.atlassian.net/browse/LPD-25421

Resending this PR to add tests as requested here: https://github.com/brianchandotcom/liferay-portal/pull/150348#issuecomment-2112104684

Since we do not have a post method available, I had to make the test as close of an equivalent as I could.

### CODE REVIEW

**Comment by SamZiemer** [comment]
ci:test:relevant

**Review by jkappler — APPROVED** [review]

**Comment by jkappler** [comment]
looks good @SamZiemer!!

**Comment by jkappler** [comment]
ci:forward

**Comment by SamZiemer** [comment]
@jkappler I've made another change to allow the one failing test to pass as well as some changes that will be useful once I add the bulk get endpoints.

**Comment by SamZiemer** [comment]
ci:test:sf

**Comment by SamZiemer** [comment]
ci:test:relevant

File: modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/DocumentShortcutResourceTest.java [hunk 1]
```diff
@@ -48,8 +48,26 @@ public class DocumentShortcutResourceTest
...
+		throws Exception {
+
+		return _addDocumentShortcut();
+	}
+
+	private DocumentShortcut _addDocumentShortcut() throws Exception {
+		return _addDocumentShortcut(testGroup);
+	}
+
+	private DocumentShortcut _addDocumentShortcut(Group group)
```
[the remark this answers is not in the document]
Comment (SamZiemer) [reply modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/DocumentShortcutResourceTest.java:67]: I've made that change, and I also made _fileshortcut no longer be a class level variable as it was not used anywhere outside the _addDocument method.

**Review by jkappler — CHANGES_REQUESTED** [review]
@SamZiemer, Brian will most likely clode this again, since there are no tests, can you add a very simple one please? 

Also, maybe to have it tidier, squash the las 4 commits? The ones that are for the tests.

Thanks

**Comment by SamZiemer** [comment]
ci:test:sf

**Comment by SamZiemer** [comment]
ci:test:relevant

**Review by jkappler — APPROVED** [review]

**Comment by jkappler** [comment]
ci:forward

**Comment by SamZiemer** [comment]
ci:test:relevant

**Comment by SamZiemer** [comment]
ci:test:relevant

**Comment by SamZiemer** [comment]
ci:test:sf

**Comment by SamZiemer** [comment]
ci:reevaluate:1337401_6011

**Comment by jonathanmccann** [comment]
Workspaces failure seems to be unrelated and is failing in other pulls as well.

**Comment by SamZiemer** [comment]
Thanks for checking that. I'll force forward this pull.

**Comment by SamZiemer** [comment]
ci:forward:force

**Comment by SamZiemer** [comment]
ci:forward:force

**Comment by ambrinchaudhary** [comment]
ci:forward:force

**Comment by ambrinchaudhary** [comment]
This rest builder failures are related to this pull @SamZiemer, must fix it before forwarding the pull

**Comment by SamZiemer** [comment]
ci:test:stable

**Comment by SamZiemer** [comment]
ci:forward:force

**Comment by SamZiemer** [comment]
ci:forward:force

**Comment by SamZiemer** [comment]
ci:forward:force

**Comment by jkappler** [comment]
ci:forward:force

**Comment by SamZiemer** [comment]
ci:stop

**Pull request closed without merging**